model/lists/Service/ListUpdater.php

```diff
@@ -18,67 +18,112 @@
  * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
  */
 
+declare(strict_types=1);
+
 namespace oat\taoBackOffice\model\lists\Service;
 
 use oat\tao\model\Lists\Business\Domain\Value;
 use oat\tao\model\Lists\Business\Domain\ValueCollection;
 use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
 use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
 use oat\tao\model\Lists\Business\Service\ValueCollectionService;
-use oat\taoBackOffice\model\lists\Contract\ListElementsUpdaterInterface;
+use oat\tao\model\Lists\DataAccess\Repository\ValueConflictException;
+use oat\taoBackOffice\model\lists\Contract\ListUpdaterInterface;
+use oat\taoBackOffice\model\lists\ListService;
+use Psr\Http\Message\ServerRequestInterface;
 use core_kernel_classes_Class;
 use tao_helpers_Uri;
+use BadFunctionCallException;
+use OverflowException;
 
-class ListElementsUpdater implements ListElementsUpdaterInterface
+class ListUpdater implements ListUpdaterInterface
 {
-    private const HUGE_LIST_MIN_ITEMS = 1000;
-    private const HUGE_LIST_MAX_MEMORY_MB = 500;
-
     /** @var ValueCollectionService */
     private $valueCollectionService;
 
-    public function __construct(ValueCollectionService $valueCollectionService)
+    /** @var ListService */
+    private $listService;
+
+    public function __construct(
+        ValueCollectionService $valueCollectionService,
+        ListService $listService
+    )
     {
         $this->valueCollectionService = $valueCollectionService;
+        $this->listService = $listService;
     }
 
-    public function setListElements(
+    /**
+     * @throws BadFunctionCallException if the payload contains too many items
+     * @throws OverflowException if the list exceeds the allowed number of items
+     * @throws ValueConflictException if element URIs are non-unique
+     */
+    public function updateByRequest(ServerRequestInterface $request): bool
```

Comment:
Do not you think it is better to have a single way of addressing errors, which would be exception. Now this can either throw exception or return false in case of error.

Since this is a `command` method, it should return void.

```suggestion
    public function updateByRequest(ServerRequestInterface $request): void
```

Reply:
Changed: 125ab2334b56311df7e3215f13d562cd369b75f7